instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/net/InetSocketAddressNetRequestAttributesExtractor.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.instrumenter.net;
+
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+/**
+ * Extractor of <a
+ * href="https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/span-general.md#general-network-connection-attributes">Network
+ * attributes</a> from a {@link InetSocketAddress}. Most network libraries will provide access to a
+ * {@link InetSocketAddress} so this is a convenient alternative to {@link
+ * NetRequestAttributesExtractor}. There is no meaning to implement both in the same
+ * instrumentation.
+ *
+ * <p>This extractor implementation uses both {@code request} and {@code response} to extract
+ * network peer information. If those attributes are available in the request, it is recommended to
+ * use {@link InetSocketAddressNetAttributesExtractor} instead.
+ */
+public abstract class InetSocketAddressNetRequestAttributesExtractor<REQUEST, RESPONSE>
```

Comment:
Didn't realize this would be needed already :)

I have considered an approach where there is only one extractor, and `@Nullable RESPONSE response` - if it's available when response is null (onStart), then it's gotten from request, otherwise gotten when response isn't null (onEnd). The advantage of two classes is it's easier to implement them. The disadvantage is having cognitive load deciding which class to use. I think I'm leaning towards the separate classes actually but wanted to get your thoughts too.